src/eradiate/scenes/spectra/_core.py

```diff
@@ -217,3 +241,21 @@ def integral(self, wmin: pint.Quantity, wmax: pint.Quantity) -> pint.Quantity:
             Computed integral value.
         """
         raise NotImplementedError
+
+    @singledispatchmethod
+    def select_in(self, spectral_set) -> Spectrum:
+        raise NotImplementedError
+
+    @select_in.register(WavelengthSet)
+    def _(self, spectral_set) -> Spectrum:
+        return self.select_in_wavelength_set(spectral_set)
+
+    @select_in.register(BinSet)
+    def _(self, spectral_set) -> Spectrum:
+        return self.select_in_bin_set(spectral_set)
+
+    def select_in_wavelength_set(self, wset) -> Spectrum:
```

Comment:
Docstring is missing (please mention that the default implementation raises `NotImplementedError`)